rfcs/0081-dynmical-config-for-stream-job.md

```diff
@@ -0,0 +1,81 @@
+---
+feature: Dynamic Config For Stream Job
+authors:
+  - "st1page"
+start_date: "2023/12/07"
+---
+
+# Dynamic Config For Stream Job
+
+Thanks all the people took part in the discussion of https://github.com/risingwavelabs/risingwave/issues/11929 and implementation of `stream_rate_limit`. The RFC is inspired by them.
+
+## Background
+Currently, we have lots of config, switch and parameter for the streaming execution. They often used for performance optimization for different workload. But they distributed different places currently.
```

Comment:
What part of the following will be unified (replaced) with the new mechanism?

Reply:
I will rewrite the part later

Reply:
I think all the cases in the example are able to be implemented as the new mechanism, but we can implement them on-demand

Reply:
For constant value I think that's easy and we can delay. 

For others that are already configurable by some mechanisms now, I'm mainly interested in migration/deprecation plan. e.g., for `OverWindowCachePolicy`, on one hand it's persisted in the plan, and on the other hand it's controlled by a session variable (but not documented yet? so less likely a large concern). 

I think unify them with the new mechanism can reduce users' mental burden. Just want to see these discussed.

Reply:
Good question! I will think and add the methods to migration for these different cases.